lib/spack/spack/test/cmd/external.py

```diff
@@ -38,6 +40,7 @@ def test_find_external_single_package(mock_executable, executables_found):
     assert single_entry.spec == Spec('cmake@1.foo')
 
 
+@pytest.mark.skipif(sys.platform == 'win32', reason="Not yet implemented on windows")
```

Comment:
(question) I"m curious what goes wrong with this test. In general it looks like all the tests which invoke `by_executable` fail on Windows.

Reply:
Seems as though `by_executable` returns an empty list and the call to next hits an iteration error. Will add to list of tests to restore.